Objects/abstract.c

```diff
@@ -881,7 +881,7 @@ binary_op1(PyObject *v, PyObject *w, const int op_slot
 
     if (slotv) {
         PyObject *x;
-        if (slotw && PyType_IsSubtype(Py_TYPE(w), Py_TYPE(v))) {
+        if (slotw && PyObject_TypeCheck(w, Py_TYPE(v))) {
```

Comment:
It is a pessimization. Types of `v` and `w` are always differetent if `slotw != NULL`. See lines 872-880.

Reply:
good point, thanks for flagging it - reverted this change